### PR TITLE
MAYA-110702 - Prim AE template cleanup Xformable section

### PR DESCRIPTION
**Pull request by seando-adsk**
#### MAYA-110702 - Prim AE template cleanup Xformable section
* Cleaned up Xformable section as per design specs.
* Added AE template strings to resource.

### CODE REVIEW

File: lib/mayaUsd/resources/ae/usdschemabase/ae_template.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
 import mayaUsd.ufe as mayaUsdUfe
 import mayaUsd.lib as mayaUsdLib
 import maya.internal.common.ufe_ae.template as ufeAeTemplate
+from mayaUsdLibRegisterStrings import getMayaUsdLibString
```
Comment (seando-adsk) [inline lib/mayaUsd/resources/ae/usdschemabase/ae_template.py:26]: I took this opportunity to put the template strings in the resource.

File: lib/mayaUsd/resources/ae/usdschemabase/ae_template.py [hunk 2]
```diff
@@ -415,6 +418,10 @@ def connectionsCustomControlCreator(aeTemplate, c):
         return None
 
 def arrayCustomControlCreator(aeTemplate, c):
+    # Note: UsdGeom.Tokens.xformOpOrder is a exception we want it to display normally.
+    if c == UsdGeom.Tokens.xformOpOrder:
+        return None
```
Comment (seando-adsk) [inline lib/mayaUsd/resources/ae/usdschemabase/ae_template.py:423]: Keep xformOpOrder as exception in that we don't want it to display as an array custom control. But just as a generic string attribute (so you can see the string value).

File: lib/mayaUsd/resources/ae/usdschemabase/ae_template.py [hunk 3]
```diff
@@ -735,14 +742,13 @@ def createTransformAttributesSection(self, sectionName, attrsToAdd):
...
+            self.addControls(xformOpOrderNames)
 
-                # Get the remainder of the xformOps and add them in an Unused section.
-                xformOpUnusedNames = fnmatch.filter(allAttrs, 'xformOp:*')
-                xformOpUnusedNames = [ele for ele in xformOpUnusedNames if ele not in xformOpOrderNames]
-                self.createSection('Unused Transform Attributes', xformOpUnusedNames, collapse=True)
+            # Get the remainder of the xformOps and add them in an Unused section.
+            xformOpUnusedNames = fnmatch.filter(allAttrs, 'xformOp:*')
+            xformOpUnusedNames = [ele for ele in xformOpUnusedNames if ele not in xformOpOrderNames]
+            self.createSection(getMayaUsdLibString('kLabelUnusedTransformAttrs'), xformOpUnusedNames, collapse=True)
```
Comment (seando-adsk) [inline lib/mayaUsd/resources/ae/usdschemabase/ae_template.py:751]: Cleanup this section as outlined in JIRA task.

File: lib/mayaUsd/resources/ae/usdschemabase/ae_template.py [hunk 4]
```diff
@@ -878,17 +884,14 @@ def buildUI(self):
...
         if not self.showArrayAttributes:
             for attrName in self.attrS.attributeNames:
                 if self.isArrayAttribute(attrName):
                     self.suppress(attrName)
 
     def isArrayAttribute(self, attrName):
-        # Note: UsdGeom.Tokens.xformOpOrder is a exception,
-        #       we are not considering this attribute as an array.
-        if self.attrS.attributeType(attrName) == ufe.Attribute.kGeneric and \
-           attrName != UsdGeom.Tokens.xformOpOrder:
```
Comment (seando-adsk) [inline lib/mayaUsd/resources/ae/usdschemabase/ae_template.py:891]: We no longer want xformOpOrder to be an exception for the "Show Array Attributes" toggle.

**Review by JGamache-autodesk — APPROVED** [review]

**Pull request merged**